cmd/awsroles.go

```diff
@@ -0,0 +1,224 @@
+//
+// Copyright (c) 2021 SSH Communications Security Inc.
+//
+// All rights reserved.
+//
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/SSHcom/privx-sdk-go/api/rolestore"
+	"github.com/spf13/cobra"
+)
+
+type awsRoleOptions struct {
+	awsRoleID string
+	refresh   bool
+}
+
+func init() {
+	rootCmd.AddCommand(awsRoleListCmd())
+}
+
+//
+//
+func awsRoleListCmd() *cobra.Command {
+	options := awsRoleOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "aws-roles",
+		Short: "List and manage AWS role links",
+		Long:  `List and manage AWS role links`,
+		Example: `
+	privx-cli aws-roles [access flags] --refresh=true
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return awsRoleList(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&options.refresh, "refresh", false, "refresh the AWS roles from AWS directories before fetching")
+
+	cmd.AddCommand(awsRoleShowCmd())
+	cmd.AddCommand(awsRoleDeleteCmd())
+	cmd.AddCommand(awsRoleUpdateCmd())
+	cmd.AddCommand(linkedRoleListCmd())
+
+	return cmd
+}
+
+func awsRoleList(options awsRoleOptions) error {
+	api := rolestore.New(curl())
+
+	roles, err := api.AWSRoleLinks(options.refresh)
+	if err != nil {
+		return err
+	}
+
+	return stdout(roles)
+}
+
+//
+//
+func awsRoleShowCmd() *cobra.Command {
+	options := awsRoleOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "show",
+		Short: "Get AWS role by ID",
+		Long:  `Get AWS role by ID. AWS role ID's are separated by commas when using multiple values, see example`,
+		Example: `
+	privx-cli aws-roles show [access flags] --id <AWS-ROLE-ID>,<AWS-ROLE-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return awsRoleShow(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.awsRoleID, "id", "", "AWS role ID")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func awsRoleShow(options awsRoleOptions) error {
```

Comment:
function returns a list of objects, "show" is consistently used for showing a single object